x-pack/legacy/plugins/apm/common/runtime_types/agent_configuration_intake_rt/index.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { transactionSampleRateRt } from '../transaction_sample_rate_rt';
+import { transactionMaxSpansRt } from '../transaction_max_spans_rt';
+
+export const serviceRt = t.intersection([
+  t.partial({ name: t.string }),
+  t.partial({ environment: t.string })
+]);
+
+export const agentConfigurationIntakeRt = t.intersection([
+  t.partial({ agent_name: t.string }),
+  t.type({ service: serviceRt }),
```

Comment:
should `service` be required here? It feels like the wrong type. We have:

```ts
{
	service: {
		name?: string;
		environment?: string;
	}
}
```

but perhaps we need:
```ts
{
	service?: { name: string } | { environment: string } | { name: string; environment: string }
}
```

Reply:
I agree - it looks pretty verbose to require the user to send `service: {}`. But it seems almost too easy to overwrite the "all/all" config by forgetting to supply the `service` part.